src/java.desktop/share/native/libawt/java2d/loops/Blit.c

```diff
@@ -74,19 +74,16 @@ Java_sun_java2d_loops_Blit_Blit
 
     srcInfo.bounds.x1 = srcx;
     srcInfo.bounds.y1 = srcy;
-    if (UNSAFE_TO_ADD(srcx, width) ||
-        UNSAFE_TO_ADD(srcy, height) ||
-        UNSAFE_TO_ADD(dstx, width) ||
-        UNSAFE_TO_ADD(dsty, height)) {
-        return;
-    }
-
-    srcInfo.bounds.x2 = srcx + width;
-    srcInfo.bounds.y2 = srcy + height;
+    srcInfo.bounds.x2 = UNSAFE_TO_ADD(srcx, width)
+                        ? clipInfo.bounds.x2 : (srcx + width);
+    srcInfo.bounds.y2 = UNSAFE_TO_ADD(srcy, height)
+                        ? clipInfo.bounds.y2 : (srcy + height);
     dstInfo.bounds.x1 = dstx;
     dstInfo.bounds.y1 = dsty;
-    dstInfo.bounds.x2 = dstx + width;
-    dstInfo.bounds.y2 = dsty + height;
+    dstInfo.bounds.x2 = UNSAFE_TO_ADD(dstx, width)
+                        ? clipInfo.bounds.x2 : (dstx + width);
+    dstInfo.bounds.y2 = UNSAFE_TO_ADD(dsty, height)
+                        ? clipInfo.bounds.y2 : (dsty + height);
     if (UNSAFE_TO_SUB(srcx, dstx) ||
         UNSAFE_TO_SUB(srcy, dsty)) {
```

Comment:
we also should check this code, is that correct to simple exit, or we should select the distance we can handle as well?